File: Python/Program/Step.py
from BasicMove import BasicMove

class Step(BasicMove):
    def __init__(self, name, duration, difficulty, intensity):
        super().__init__(name, duration, difficulty)
        self.__stepIntensity = intensity
    
    def setStepIntensity(self, intensity):
        self.__stepIntensity = intensity
    def getStepIntensity(self):
        return self.__stepIntensity

File: CPP/Program/Step.cpp
#ifndef STEP_CPP
#define STEP_CPP

#include <string>
#include "BasicMove.cpp"
using namespace std;

class Step : public BasicMove{
    private:
        int stepIntensity; // 1 = low, 2 = medium, 3 = high
    public:
        Step(string name, int duration, string difficulty, int intensity) : BasicMove(name, duration, difficulty){
            this->stepIntensity = intensity;
        }

        void setStepIntensity(int intensity){
            this->stepIntensity = intensity;
        }
        int getStepIntensity(){
            return this->stepIntensity;
        }

        ~Step(){}
};

#endif